From: Andrew Morton <akpm@linux-foundation.org>
Subject: mm-vmscan-fix-comment-for-current_may_throttle-fix

tweak grammar

Cc: Miaohe Lin <linmiaohe@huawei.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---


--- a/mm/vmscan.c~mm-vmscan-fix-comment-for-current_may_throttle-fix
+++ a/mm/vmscan.c
@@ -2332,9 +2332,9 @@ static unsigned int move_pages_to_lru(st
 }
 
 /*
- * If a kernel thread (such as nfsd for loop-back mounts) services
- * a backing device by writing to the page cache it sets PF_LOCAL_THROTTLE.
- * In that case we should not throttle it otherwise it is safe to do so.
+ * If a kernel thread (such as nfsd for loop-back mounts) services a backing
+ * device by writing to the page cache it sets PF_LOCAL_THROTTLE. In this case
+ * we should not throttle.  Otherwise it is safe to do so.
  */
 static int current_may_throttle(void)
 {
_
